app/views/homepage/index.haml

```diff
@@ -96,19 +96,19 @@
               .toggle-header-container
                 .toggle-header
                   - if selected_category
-                    = selected_category.display_name(I18n.locale)
+                    = @category_display_names[selected_category.id]
```

Comment:
@bladealslayer 
This is solved by RoR. You do not need your own implementation. https://guides.rubyonrails.org/caching_with_rails.html#fragment-caching

Reply:
@aivils I tried that, but the issue is then there will be one fragment cached for each state of the navigation menu - i.e. when user selects different category, different item will be created in the cache, so the cache isn't that useful. Imagine marketplace with 100+ categories and multiple languages - for a cold cache, each click on the category menu will need to create new cached menu fragment. Do you know some way to take care of that issue with fragment cache?

Reply:
@bladealslayer 
From my point of view keeping code closer to genuine RoR is more efficient in the long term. RoR solves same troubles we solve. Here you should make precise measurements to decide what is more correct.

Reply:
Indeed. However, I couldn't get satisfactory result. Imagine the following scenario (which is actually happening): marketplace with 300 categories and 20+ languages, where a web crawler starts crawling. As I said above, if the menu is cached as fragment, separate cache entries will exist for each combination of locale and `current_category` that are there. So if a crawler / user clicks through all categories one by one, each consequent request will not take advantage of the menu cached on the previous click.